src/components/NavBar/ChainSelector.tsx

```diff
@@ -74,6 +44,18 @@ export const ChainSwitcher = ({ leftAlign }: ChainSwitcherProps) => {
   const selectChain = useSelectChain()
   useSyncChainQuery()
 
+  const [pendingChainId, setPendingChainId] = useState<SupportedChainId | undefined>(undefined)
+
+  const onSelectChain = useCallback(
+    async (targetChainId: SupportedChainId) => {
+      setPendingChainId(targetChainId)
+      await selectChain(targetChainId)
+      setPendingChainId(undefined)
+      toggleOpen()
```

Comment:
Can we add a check here to make sure the Selector is open before toggling? Currently if I close the modal before approving in my wallet, the modal reopens.

Reply:
ended up changing out of the reducer to solve it